src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
 }
```

Comment:
It works) Here is the solution from me, maybe it will be useful for you to understand) 

function calculateAverageAgeDiff(people, onlyWithSon) {
  const children = people.filter(child =>
    (!onlyWithSon || child.sex === 'm') && child.mother
  );

  const totalAgeDiff = children.reduce((sum, child) => {
    const mother = people.find(person => person.name === child.mother);
    return sum + (child.born - mother.born);
  }, 0);

  return totalAgeDiff / children.length;
}